runtime/src/main/java/org/corfudb/util/Utils.java

```diff
@@ -465,6 +465,28 @@ public static void printLogAnatomy(CorfuRuntime runtime, ILogData logData) {
     }
 
 
+    /**
+     * Given a set of request tails, we aggregate them and maintain
+     * the greatest address per stream and the greatest tail over
+     * all responses.
+     * @param responses a set of tail responses
+     * @return An max-aggregation of all tails
+     */
+    static Tails getTails(Set<TailsResponse> responses) {
+        long globalTail = Address.NON_ADDRESS;
+        Map<UUID, Long> globalStreamTails = new HashMap<>();
+
+        for (TailsResponse res : responses) {
+            globalTail = Math.max(globalTail, res.getLogTail());
+
+            for (Map.Entry<UUID, Long> stream : res.getStreamTails().entrySet()) {
+                long streamTail = globalStreamTails.getOrDefault(stream.getKey(), Address.NON_ADDRESS);
+                globalStreamTails.put(stream.getKey(), Math.max(streamTail, stream.getValue()));
+            }
+        }
+        return new Tails(globalTail, globalStreamTails);
+    }
+
```

Comment:
Why we have this method in Utils class?

Reply:
There's two code paths, one that needs to compute the tails given a layout and another that just calls it from the address space that uses any "current" layout. The two paths should be merged, but not in this pr.